src/bootstrap/test.rs

```diff
@@ -1124,7 +1124,7 @@ impl Step for Compiletest {
         flags.push(builder.config.cmd.rustc_args().join(" "));
 
         if let Some(linker) = builder.linker(target) {
-            cmd.arg("--linker").arg(linker);
+            cmd.arg(format!("-Clinker={}", linker.display()));
```

Comment:
Same as before, we want to avoid `display()` if at all possible as it's lossy, we can pass two args here just fine

Reply:
ought not have updated this file at all as it is not calling rustdoc
resulting in failed test